xwiki-platform-core/xwiki-platform-index/xwiki-platform-index-api/src/main/java/org/xwiki/index/TaskConsumer.java

```diff
@@ -17,36 +17,30 @@
  * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
  * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
  */
-package org.xwiki.mentions.internal;
-
-import java.util.concurrent.BlockingQueue;
+package org.xwiki.index;
 
+import org.suigeneris.jrcs.rcs.Version;
 import org.xwiki.component.annotation.Role;
-import org.xwiki.mentions.MentionException;
-import org.xwiki.mentions.internal.async.MentionsData;
+import org.xwiki.stability.Unstable;
 
 /**
- * Blocking queue provider.
+ * Provide the operation to consume a task.
  *
  * @version $Id$
- * @since 12.6
+ * @since 14.1RC1
  */
 @Role
-public interface MentionsBlockingQueueProvider
+@Unstable
+public interface TaskConsumer
 {
     /**
-     * Initialize a blocking queue.
-     *
-     * @return the blocking queue
-     * @throws MentionException if an error occurs during the initialization of the queue
-     */
-    BlockingQueue<MentionsData> initBlockingQueue() throws MentionException;
-
-    /**
-     * Close the queue.
+     * Consume a task.
      *
-     * @since 12.6.1
-     * @since 12.7RC1
+     * @param wikiId the identifier of the wiki where the task is to be executed
+     * @param docName the reference of the document to analyze
+     * @param version the version of the document to analyze
+     * @param author the reference of the user to use to perform the analyzis
+     * @throws IndexException in case of error during the execution of the task
      */
-    void closeQueue();
+    void consume(String wikiId, String docName, Version version, String author) throws IndexException;
```

Comment:
Author might actually be useless and can be derived from the `XWikiDocument` properties